amd/src/courseformat/courseindex/section.js

```diff
@@ -156,7 +196,19 @@ export default class Component extends ComponentBase {
             while (target.levelsan > origin.levelsan) {
                 target = state.section.get(this.course.sectionlist[target.number - 1]);
             }
-            this.reactive.dispatch('fmtSectionMove', origin, target);
+            const moveDirection = Math.sign(target.number - origin.number);
+            let targetCall = target;
+            if (moveDirection > 0 && target.indexcollapsed) {
+                let targetChild = target;
+                while (this.course.sectionlist.length > targetChild.number + 1
+                && state.section.get(this.course.sectionlist[targetChild.number + 1]).levelsan > target.levelsan) {
+                    targetChild = state.section.get(this.course.sectionlist[targetChild.number + 1]);
+                    if (targetChild.levelsan >= origin.levelsan) {
```

Comment:
I think I got this backwards.
```suggestion
                    if (targetChild.levelsan <= origin.levelsan) {
```